### PR TITLE
Update api-async.mdx

### PR DESCRIPTION
**Pull request by lukaselmer**
I expected `waitFor` to wait for a condition described in the `callback`. For example, I expected

```js
await waitFor(() => false)
```

to wait forever / until the timeout is reached. A better documentation would have helped to realize this quicker.

### CODE REVIEW

File: docs/dom-testing-library/api-async.mdx [hunk 1]
```diff
@@ -62,6 +62,11 @@ options.
 This can be useful if you have a unit test that mocks API calls and you need to
 wait for your mock promises to all resolve.
 
+The callback passed to `waitFor` _must throw an error_ for as long as it should
```
[the remark this answers is not in the document]
Comment (lukaselmer) [reply docs/dom-testing-library/api-async.mdx]: Thanks for the review. Yes, that's better. I've extended it to also state that the error must be thrown by an `expect(...)`.

[the remark this answers is not in the document]
Comment (lukaselmer) [reply docs/dom-testing-library/api-async.mdx:67]: Thank you for the review. Yes, that's better! I've updated the commit to reflect your suggestion.

File: docs/dom-testing-library/api-async.mdx [hunk 2]
```diff
@@ -62,6 +64,11 @@ options.
...
+be waited for. The error must be of the same form as failed `expect` errors.
+The simplest way to achieve that is to write the condition as an expectation
+(see example above).
+
```
Comment (timdeschryver) [inline docs/dom-testing-library/api-async.mdx:71]: I like the change, but I think we should also revert this change and remove the paragraph.

```suggestion
```

**Comment by timdeschryver** [comment]
@all-contributors please add @lukaselmer  for docs
